### ISSUE TITLE
Fix false negatives for template literal declarations in declaration-empty-line-before

### ISSUE THREAD

**Issue by hudochenkov** [stylelint/stylelint#4003]
> Clearly describe the bug

Warning is not shown for `declaration-empty-line-before` in styled component.

> Which rule, if any, is the bug related to?

`declaration-empty-line-before`

> What CSS is needed to reproduce the bug?

```js
const Separator = styled.div`
  margin-bottom: 20px;
  
  border: 0;
`;
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "declaration-empty-line-before": ["never"]
  }
}
```

> Which version of stylelint are you using?

`9.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "**/*.styled.js"`. Also in [demo](https://stylelint.io/demo/)

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to CSS-in-JS (styled components).

> What did you expect to happen?

Warnings to be flagged for `border`.

> What actually happened (e.g. what warnings or errors did you get)?

No warnings.

**Comment by hudochenkov** [comment]
The issue in 

https://github.com/stylelint/stylelint/blob/ad4669882dc80dea68a001cdf5d0dfb61611cc0f/lib/rules/declaration-empty-line-before/index.js#L57-L59

https://github.com/stylelint/stylelint/blob/ad4669882dc80dea68a001cdf5d0dfb61611cc0f/lib/utils/isStandardSyntaxDeclaration.js#L13-L15

In `postcss-jsx` parser properties are children of `root`.

**Comment by jeddy3** [comment]
> In postcss-jsx parser properties are children of root.

I believe that's there because we don't want this rule to pick up on "declarations" outside of rules e.g. dollar variables. We can either work out another heuristic for this or change `postcss-jsx` behaviour so these declarations are children of a rule node with selector "div".

**Comment by evilebottnawi** [comment]
I think fix `postcss-jsx` behavior is right way

**Comment by hudochenkov** [comment]
@evilebottnawi what this change could be? If `postcss-jsx` will wrap styles in `div` on parse, then it will introduce false positives for `selector-max-type: 0`, for example.

**Comment by evilebottnawi** [comment]
@hudochenkov hm, maybe we can add `context.isStyled` (same we can do for markdown/html/etc if needed) and handle this cases?

**Comment by hudochenkov** [comment]
I was reviewing PRs and found https://github.com/stylelint/stylelint/pull/3933 which should fix this problem. This issue renders to be a duplicate of https://github.com/stylelint/stylelint/issues/3930.
